### ISSUE TITLE
Data Elements have same idx value

### ISSUE THREAD

**Issue by jamesrwelch** [MauroDataMapper/mdm-core#234]
The data elements in the bootstrapped data model `Complex Data Model`, class `Content` have the same `idx` value. 

This may also be an issue in creating a new version - an example in the NHSD test instance has a class in v0.0.1 of the model with idx values 0 and 1, but in the v0.0.2 model, the idx for those equivalent elements is 1 and 1.

Editing the description of an element does not fix the idx of that element.

**Comment by olliefreeman** [comment]
okay, so im finding an issue on DCs inside bootstrapped DMs due to the way we build and save the DM. however for a branched model/copy there should be no way for this to be an issue as a DM or DC without an id will call a full sort on its children, so i'm not sure what happened with your v0.0.2

**Comment by olliefreeman** [comment]
@jamesrwelch do we want to accept a "startup hit" by adding a migration script which checks and sorts all the idx values? or just leave as is?

**Comment by jamesrwelch** [comment]
@olliefreeman I'm afraid I think we do want to take that hit, otherwise existing models with the issue will continue to have the same problem, and this might get reported again.  I'll have a quick play and try to work out why the 0.0.2 model is in that strange state

**Comment by olliefreeman** [comment]
This is fixed through migration in v5, however leaving it open as we need to see if it strikes again and if so where and how

**Comment by jamesrwelch** [comment]
@joe-crawford Does this sound like something fixed by #343?

**Comment by joe-crawford** [comment]
Yes this should be fixed by the change to always sort child ModelItems on saving in #343.

**Comment by joe-crawford** [comment]
There's a test in that PR against the bootstrapped Complex Data Model, the issue was the ModelItems weren't being sorted when multiple new ModelItems were added to the DataModel across multiple different saves.
